src/adapter/etl-adapter-google-sheet/tests/Flow/ETL/Adapter/GoogleSheet/Tests/Unit/GoogleSheetExtractorTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\GoogleSheet\Tests\Unit;
+
+use Flow\ETL\Adapter\GoogleSheet\Columns;
+use Flow\ETL\Adapter\GoogleSheet\GoogleSheetExtractor;
+use Flow\ETL\ConfigBuilder;
+use Flow\ETL\FlowContext;
+use Flow\ETL\Row\Entry\ArrayEntry;
+use Flow\ETL\Rows;
+use Google\Service\Sheets;
+use Google\Service\Sheets\Resource\SpreadsheetsValues;
+use PHPUnit\Framework\TestCase;
+
+final class GoogleSheetExtractorTest extends TestCase
+{
+    public function test_its_stop_fetching_data_if_processed_row_count_is_less_then_last_range_end_row() : void
+    {
+        $extractor = new GoogleSheetExtractor(
```

Comment:
could you please use DSL in tests? This way we also cover the DSL with some tests

Reply:
yeah, sure

Reply:
But now  I will have to inject mock in a different way

Reply:
ah right, you are mocking it, hmm ok, then leave it as is for now